### PR TITLE
45 add new parameters to cli and refactor

### PR DESCRIPTION
**Pull request by Fr3dMer**
See issue #45, goal was:
- [x] Add patient ID and sample ID parameters
- [x] Add parameters for saving BED files
- [x] Add logic for checking both are present 
- [x] Move general error catching to CLI
- [x] Add doc_string 
- [x] Improve comments 
- [x] Update tests

### CODE REVIEW

File: test/cli_test.py [hunk 1]
```diff
@@ -8,47 +8,62 @@
...
-#-----------------------------------------------------------------------------
-    # Normal CLI
-    def test_cli1_init(self):
-        
-        cli = cli_obj.cli_obj(normal)
-        assert cli.args.panel_id == int(normal_pan[1])
-        assert cli.args.rcode == normal_rcode[1]
-        assert cli.args.human == True
+
+def test_cli_obj_valid_args():
```
Comment (AmyGrimwood) [inline test/cli_test.py:12]: Does this need a docstring?

File: test/cli_test.py [hunk 2]
```diff
@@ -8,47 +8,62 @@
...
-        with pytest.raises(SystemExit):
-            cli_obj.cli_obj(no_rcode)
+def test_cli_obj_invalid_bed_paths_37():
+    # Test case for invalid bed37 file paths
+    args = ["-p", "1", "-r", "rcode", "-b37", "/invalid/path/bed37"]
+    
+    with pytest.raises(SystemExit):
+        cli = cli_obj.cli_obj(args)
 
+def test_cli_obj_invalid_bed_paths_38():
```
Comment (AmyGrimwood) [inline test/cli_test.py:49]: Docstring and 2 lines before defining function?

File: test/cli_test.py [hunk 3]
```diff
@@ -8,47 +8,62 @@
...
+def test_cli_obj_no_sampleid():
+    # Test case for not providing sampple_id wiht rcode
+    args = ["-pid", "patientID"]
+    
+    with pytest.raises(SystemExit):
+        cli = cli_obj.cli_obj(args)
 
-        cli = cli_obj.cli_obj(normal[0:-1])
-        assert cli.args.human == False
+def test_cli_obj_no_patientid():
```
Comment (AmyGrimwood) [inline test/cli_test.py:63]: Docstring and 2 lines before defining function?

File: test/cli_test.py [hunk 4]
```diff
@@ -8,47 +8,62 @@
...
+def test_cli_obj_invalid_bed_paths_38():
+    # Test case for invalid bed37 file paths
+    args = ["-p", "1", "-r", "rcode", "-b38", "/invalid/path/bed38"]
+    
+    with pytest.raises(SystemExit):
+        cli = cli_obj.cli_obj(args)
 
-    # Finally test Human bool flag 
-    def test_cli3_init(self):
+def test_cli_obj_no_sampleid():
```
Comment (AmyGrimwood) [inline test/cli_test.py:56]: Docstring and 2 lines before defining function?

**Review by AmyGrimwood — CHANGES_REQUESTED** [review]
A couple potential PEP8 issues. Let me know if your planning to address PEP8 in a later issue rather than in this pull, and I'll approve

**Comment by AbiHaddon** [comment]
I agree with Amy's review. Mainly require doc string. If you wish to address this at in a later pull i'll approve

**Review by AbiHaddon — CHANGES_REQUESTED** [review]

**Comment by AbiHaddon** [comment]
> A couple potential PEP8 issues. Let me know if your planning to address PEP8 in a later issue rather than in this pull, and I'll approve

image
I agree with Amy. I got the above pep8 issues. As stated let us know if you wish to address this later and we'll approve.

**Comment by Fr3dMer** [comment]
Will fix the PEP-8 issues. A lot of the docstring issues are for the tests, which aren't public facing so do the tests need docstrings? Happy to add a comment describing what each function is testing instead   
>[Write docstrings for all public modules, functions, classes, and methods. Docstrings are not necessary for non-public methods, but you should have a comment that describes what the method does. This comment should appear after the def line. ](https://peps.python.org/pep-0008/#documentation-strings)

File: src/cli.py [hunk 5]
```diff
@@ -4,27 +4,102 @@
...
+            self.parser.error(error_msg)
+
+        # Check dirs supplied exists
+        if (type(self.args.bed37) == str):
+            
+            if not (os.path.isdir(self.args.bed37)):
+                error_msg = 'The path you provided for bed37 ' \
+                            + 'does not exist'
+                self.parser.error(error_msg)
+
```
[the remark this answers is not in the document]
Comment (Fr3dMer) [reply src/cli.py:85]: Comment added

**Review by AmyGrimwood — APPROVED** [review]
looks good, thank you for resolving the comments we had

**Pull request merged**